### PR TITLE
[GUI] Settings widget, expand/collapse scroll area.

### PR DESCRIPTION
**Pull request by furszy**
Solving #1078.

### CODE REVIEW

File: src/qt/pivx/settings/forms/settingswidget.ui [hunk 1]
```diff
@@ -56,37 +56,51 @@
...
+           <property name="rightMargin">
+            <number>20</number>
+           </property>
+           <property name="bottomMargin">
+            <number>0</number>
+           </property>
+           <item>
+            <widget class="QLabel" name="labelTitle">
+             <property name="text">
+              <string>TextLabel</string>
```
Comment (Fuzzbawls) [inline src/qt/pivx/settings/forms/settingswidget.ui]: nit: either set the static string here to `Settings` and remove the `setText()` call in the `.cpp` file

or

mark this "dummy" string as not needing translation `<string notr=true>TextLabel</string>`

File: src/qt/pivx/settings/settingswidget.cpp [hunk 2]
```diff
@@ -252,24 +266,26 @@ void SettingsWidget::onSaveOptionsClicked(){
...
-        ui->optionsButtonsWidget->setVisible(false);
-        ui->toolsButtonsWidget->setVisible(false);
-        ui->configurationButtonsWidget->setVisible(false);
-        ui->helpButtonsWidget->setVisible(false);
-        ui->pushButtonOptions->setChecked(false);
-        ui->pushButtonTools->setChecked(false);
-        ui->pushButtonConfiguration->setChecked(false);
-        ui->pushButtonHelp->setChecked(false);
-
+void SettingsWidget::selectMenu(QPushButton* btn) {
```
Comment (Fuzzbawls) [inline src/qt/pivx/settings/settingswidget.cpp]: brace on newline

File: src/qt/pivx/settings/settingswidget.cpp [hunk 3]
```diff
@@ -252,24 +266,26 @@ void SettingsWidget::onSaveOptionsClicked(){
...
+            value->setVisible(value == subMenuSelected);
+            if (key != btn) key->setChecked(false);
+        }
     } else {
-        ui->fileButtonsWidget->setVisible(false);
+        subMenuSelected->setVisible(false);
     }
 }
 
+void SettingsWidget::onFileClicked() {
```
Comment (Fuzzbawls) [inline src/qt/pivx/settings/settingswidget.cpp]: brace on newline

File: src/qt/pivx/settings/settingswidget.cpp [hunk 4]
```diff
@@ -434,7 +395,7 @@ void SettingsWidget::setMapper(){
     settingsDisplayOptionsWidget->setMapper(mapper);
 }
 
-bool SettingsWidget::openStandardDialog(QString title, QString body, QString okBtn, QString cancelBtn){
+bool SettingsWidget::openStandardDialog(const QString& title, const QString& body, const QString& okBtn, const QString& cancelBtn){
```
Comment (Fuzzbawls) [inline src/qt/pivx/settings/settingswidget.cpp]: brace on newline

File: src/qt/pivx/settings/settingswidget.cpp [hunk 5]
```diff
@@ -178,9 +175,26 @@ SettingsWidget::SettingsWidget(PIVXGUI* parent) :
...
+    ui->scrollArea->setWidgetResizable(true);
+    QSizePolicy scrollAreaPolicy = ui->scrollArea->sizePolicy();
+    scrollAreaPolicy.setVerticalStretch(1);
+    ui->scrollArea->setSizePolicy(scrollAreaPolicy);
+    QSizePolicy scrollVertPolicy = ui->scrollAreaWidgetContents->sizePolicy();
+    scrollVertPolicy.setVerticalStretch(1);
+    ui->scrollAreaWidgetContents->setSizePolicy(scrollVertPolicy);
+
+    if (window)
+        connect(window, &PIVXGUI::windowResizeEvent, this, &SettingsWidget::windowResizeEvent);
```
Comment (Fuzzbawls) [inline src/qt/pivx/settings/settingswidget.cpp]: No need to even connect here.

File: src/qt/pivx/settings/settingswidget.cpp [hunk 6]
```diff
@@ -447,3 +408,9 @@ bool SettingsWidget::openStandardDialog(QString title, QString body, QString okB
 SettingsWidget::~SettingsWidget(){
     delete ui;
 }
+
+void SettingsWidget::windowResizeEvent(QResizeEvent* event) {
```
Comment (Fuzzbawls) [inline src/qt/pivx/settings/settingswidget.cpp]: Can remove this as it isn't needed and doesn't do anything as-is

File: src/qt/pivx/settings/settingswidget.h [hunk 7]
```diff
@@ -78,8 +78,11 @@ private Q_SLOTS:
 
     void onResetAction();
     void onSaveOptionsClicked();
+
+    void windowResizeEvent(QResizeEvent* event);
```
Comment (Fuzzbawls) [inline src/qt/pivx/settings/settingswidget.h]: can remove

**Comment by furszy** [comment]
Updated solving the nits.

**Pull request merged**